### ISSUE TITLE
infer schema using spark-csv over parquet table with string columns

### ISSUE THREAD

**Issue by Tagar** [databricks/spark-csv#264]
Would be great to use spark-csv to infer schema from an arbitrary DF or a parquet table that has string columns. Or is this already possible perhaps?

**Comment by HyukjinKwon** [comment]
For me, I am not too sure if we need this but It is anyway possible. 
First of all, you might be able to do this by converting data, `DataFrame` -> `RDD[String]` -> `DataFrame` as below (if this has nested data types, this will produces wrong data):

```scala
val rdd = arbitraryDf.rdd.map(_.toSeq.mkString(","))
val df = new CsvParser().csvRdd(sqlContext, rdd)
```

**Comment by Tagar** [comment]
> map(_.toSeq.mkString(","))
wouldn't really work because some of the strings might have commas.
have a look at http://stackoverflow.com/questions/32722132/spark-data-type-guesser-udaf
thank you.

**Comment by HyukjinKwon** [comment]
@Tagar I meant to say this workaround is anyway possible in this way.
For example, we can escape from the commnas with this.
```scala
val rdd = arbitraryDf.rdd.map(_.toSeq.map(v => s""""$v"""").mkString(","))
val df = new CsvParser().csvRdd(sqlContext, rdd)
```
If this contains `"` then, we can set another separator for this and use `quote` option for reading this.
If converting `DataFrame` to `RDD[String]` can be done (filed here https://github.com/databricks/spark-csv/issues/188), we might be able to do this.

**Comment by Tagar** [comment]
I have to agree. This is possible, but looks messy and requires two passes / two conversions:
convert dataframe to... csv just to make it work, and convert that back to a dataframe.

What I am suggesting in this issue requires one pass.
We have a lot of use cases when very wide datasets have ~98% of columns being just strings,
because source systems don't give us schema.

It would be great to have a feature of schema inference over RDD of strings built into 
Spark-CSV or into Spark. Thank you.

**Comment by Tagar** [comment]
Just noticed #188 thank you @HyukjinKwon
